File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        friends = list(range(1, n + 1))
        current_position = 0
        
        while len(friends) > 1:
            to_remove = (current_position + k-1) % len(friends)
            friends.pop(to_remove)
            current_position = to_remove % len(friends)
            
        return friends[0]
       
		 
		
		
	    
        

File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number}
 */
var findTheWinner = function(n, k) {
    let friends = Array.from({ length: n }, (_, index) => index + 1);
    let current_position = 0;
    
    while (friends.length > 1) {
        let to_remove = (current_position + k - 1) % friends.length;
        friends.splice(to_remove, 1);
        current_position = to_remove % friends.length;
    }
    
    return friends[0];
    
      
};